### PR TITLE
mathematica: 12.3.1 -> 13.0.0

### PR DESCRIPTION
**Pull request by qbg**
###### Motivation for this change

https://wolfram.com/mathematica/quick-revision-history.html#v130

###### Things done

- Built on platform(s)
  -  [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of binary files: mathematica, wolfram
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Tested using one of my notebooks that uses data from a SQLite database.

Built on NixOS.

### CODE REVIEW

File: pkgs/applications/science/math/mathematica/l10ns.nix [hunk 1]
```diff
@@ -8,31 +8,34 @@ let allVersions = with lib; flip map
   # N.B. Versions in this list should be ordered from newest to oldest.
   [
     {
-      version = "12.3.1";
+      version = "13.0.0";
```
Comment (veprbl) [inline pkgs/applications/science/math/mathematica/l10ns.nix:11]: I think the point of having multiple options is to accommodate people owning licenses for different versions.

Comment (qbg) [reply pkgs/applications/science/math/mathematica/l10ns.nix:11]: The last several version updates in this file followed the same process, and it appears that `default.nix` doesn't pass in the version so it'll always will be the latest for a given language.

I'll go through the Git history and add back the last several previous versions that were removed.

**Review by FliegendeWurst — COMMENTED** [review]
Launching Mathematica 13.0 works, didn't notice any issues.

File: pkgs/applications/science/math/mathematica/l10ns.nix (hunk 1 above)
Comment (qbg) [reply pkgs/applications/science/math/mathematica/l10ns.nix:11]: I've added a commit that adds them back.

**Comment by qbg** [comment]
Pinging @herberteuler since ofborg did not.

**Review by herberteuler — APPROVED** [review]

**Pull request merged**